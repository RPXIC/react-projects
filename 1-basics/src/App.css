.flex {
    display: flex;
    flex: row;
    justify-content: space-between;
}
.center{
    text-align: center;
}