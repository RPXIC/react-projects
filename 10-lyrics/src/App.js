import React, { useState, useEffect } from 'react'
import { Form, Lyric, Info } from 'components'

const App = () => {
  const [query, setQuery] = useState({})
  const [lyric, setLyric] = useState('')
  const [info, setInfo] = useState({})

  useEffect(() => {
    if (!Object.keys(query).length) return

    (async () => {
      try {
        const { artist, title } = query
        const url = `https://api.lyrics.ovh/v1/${artist.toUpperCase()}/${title.toUpperCase()}`
        const url2 = `https://theaudiodb.com/api/v1/json/1/search.php?s=${artist.toUpperCase()}`

        const [lyricRes, infoRes] = await Promise.all([
          fetch(url),
          fetch(url2)
        ])

        const [lyric, info] = await Promise.all([
          lyricRes.json(),
          infoRes.json()
        ])

        setLyric(lyric.lyrics)
        setInfo(info.artists[0])
      } catch (error) {
        console.log(error)
      }
    })()
  }, [query])

  return (
    <>
      <Form
        setQuery={setQuery}
      />
      <div className="container mt-5">
        <div className="row">
          <div className="col-md-6">
            <Info info={info} />
          </div>
          <div className="col-md-6">
            <Lyric lyric={lyric} />
          </div>
        </div>
      </div>
    </>
  )
}

export default App
