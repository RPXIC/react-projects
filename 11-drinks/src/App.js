import React from 'react'
import { Header, Form, RecipesList } from 'components'
import { CategoriesProvider, RecipesProvider, ModalProvider } from 'context'

const App = () => {
  return (
    <>
      <CategoriesProvider>
        <RecipesProvider>
          <ModalProvider>
            <Header />
            <div className="container mt-5">
              <div className="row">
                <Form />
              </div>
              <RecipesList />
            </div>
          </ModalProvider>
        </RecipesProvider>
      </CategoriesProvider>
    </>
  )
}

export default App
