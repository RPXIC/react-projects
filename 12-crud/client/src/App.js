import React from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import { Login, Register, Projects } from 'components'
import { ProjectState, TodoState, AlertState, AuthState } from 'context'

const App = () => {
  return (
    <ProjectState>
      <TodoState>
        <AlertState>
          <AuthState>
            <Router>
              <Switch>
                <Route exact path="/" component={Login} />
                <Route exact path="/register" component={Register} />
                <Route exact path="/projects" component={Projects} />
              </Switch>
            </Router>
          </AuthState>
        </AlertState>
      </TodoState>
    </ProjectState>
  )
}

export default App
