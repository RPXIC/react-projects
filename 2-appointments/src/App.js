import React, { useState, useEffect } from 'react'
import Form from './components/Form'
import Appointment from './components/Appointment'

const App = () => {
  let InitialAppointments = JSON.parse(localStorage.getItem('appointments'))
  if (!InitialAppointments) InitialAppointments = []

  const [appointments, setAppointments] = useState(InitialAppointments)

  useEffect(() => {
    if (InitialAppointments) localStorage.setItem('appointments', JSON.stringify(appointments))
    else localStorage.setItem('appointments', JSON.stringify([]))
  }, [appointments, InitialAppointments])

  const createAppointment = appointment => {
    setAppointments([
      ...appointments,
      appointment
    ])
  }

  const deleteAppointment = id => {
    const newAppointments = appointments.filter(appointment => appointment.id !== id)
    setAppointments(newAppointments)
  }

  const title = appointments.length ? 'Manage your appointments' : 'No appointments'

  return (
    <>
      <h1>Appointments manager</h1>
      <div className="container">
        <div className="row">
          <div className="one-half column">
            <Form
              createAppointment={createAppointment}
            />
          </div>
          <div className="one-half column">
            <h2>{title}</h2>
            {appointments.map(appointment => (
              <Appointment
                key={appointment.id}
                appointment={appointment}
                deleteAppointment={deleteAppointment}
              />
            ))}
          </div>
        </div>
      </div>
    </>
  )
}

export default App
