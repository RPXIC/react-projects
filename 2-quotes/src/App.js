import React, { useState, useEffect } from 'react'
import Form from './components/Form'
import Quote from './components/Quote'

const App = () => {
  let initialQuotes = JSON.parse(localStorage.getItem('quotes'))
  if (!initialQuotes) initialQuotes = []

  const [quotes, setQuotes] = useState(initialQuotes)

  useEffect(() => {
    if (initialQuotes) localStorage.setItem('quotes', JSON.stringify(quotes))
    else localStorage.setItem('quotes', JSON.stringify([]))
  }, [quotes, initialQuotes])

  const createQuote = quote => {
    setQuotes([
      ...quotes,
      quote
    ])
  }

  const deleteQuote = id => {
    const newQuotes = quotes.filter(quote => quote.id !== id)
    setQuotes(newQuotes)
  }

  const title = quotes.length ? 'Manage your quotes' : 'No quotes'

  return (
    <>
      <h1>Quotes manager</h1>
      <div className="container">
        <div className="row">
          <div className="one-half column">
            <Form
              createQuote={createQuote}
            />
          </div>
          <div className="one-half column">
            <h2>{title}</h2>
            {quotes.map(quote => (
              <Quote
                key={quote.id}
                quote={quote}
                deleteQuote={deleteQuote}
              />
            ))}
          </div>
        </div>
      </div>
    </>
  )
}

export default App
