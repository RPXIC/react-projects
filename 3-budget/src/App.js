import React, { useState, useEffect } from 'react'
import { Question, Form, List, BudgetControl } from 'components'

const App = () => {
  const [budget, setBudget] = useState(0)
  const [remaining, setRemaining] = useState(0)
  const [showQuestion, setShowQuestion] = useState(true)
  const [expenses, setExpenses] = useState([])
  const [spending, setSpending] = useState({})
  const [createSpending, setCreateSpending] = useState(false)

  useEffect(() => {
    if (createSpending) {
      setExpenses([
        ...expenses,
        spending
      ])
      setCreateSpending(false)

      const remainingBudget = remaining - spending.quantity;

      setRemaining(remainingBudget)

      setCreateSpending(false)
    }
  }, [createSpending, expenses, remaining, spending])

  return (
    <div className="container">
      <header>
        <h1>Weekly expense</h1>
        <div className="main-content content">
          {showQuestion ?
            (
              <Question
                setBudget={setBudget}
                setRemaining={setRemaining}
                setShowQuestion={setShowQuestion}
              />
            ) : (
              <div className="row">
                <div className="one-half column">
                  <Form setSpending={setSpending} setCreateSpending={setCreateSpending} />
                </div>
                <div className="one-half column">
                  <List expenses={expenses} />
                  <BudgetControl
                    budget={budget}
                    remaining={remaining}
                  />
                </div>
              </div>
            )
          }
        </div>
      </header>
    </div>
  )
}

export default App
