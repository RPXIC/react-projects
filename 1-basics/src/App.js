import React, { useState } from 'react'
import { Header, Footer, Item, Cart } from 'components'
import './App.css'

const App = () => {
  const [items] = useState([
    { id: 1, name: 'T-Shirt ReactJS', price: 50 },
    { id: 2, name: 'T-Shirt VueJS', price: 40 },
    { id: 3, name: 'T-Shirt NodeJS', price: 30 },
    { id: 4, name: 'T-Shirt Angular', price: 20 }
  ])
  const [cart, setCart] = useState([])

  const date = new Date().getFullYear()

  return (
    <>
      <Header title="Online shop" />
      <h1>List items</h1>
      <div className="flex">
        <div>
          {items.map(item =>
            <Item
              key={item.id}
              item={item}
              cart={cart}
              items={items}
              setCart={setCart}
            />
          )}
        </div>
        <Cart cart={cart} setCart={setCart} />
      </div>
      <Footer date={date} />
    </>
  )
}

export default App
