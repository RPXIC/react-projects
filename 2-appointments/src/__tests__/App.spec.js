import React from 'react'
import { render, screen } from '@testing-library/react'
import App from '../App'
import '@testing-library/jest-dom/extend-expect'
import userEvent from '@testing-library/user-event'

it('<App /> - Render', () => {
	render(<App />)

	expect(screen.getByText('Appointments manager')).toBeInTheDocument()
	expect(screen.getByTestId('app-name').textContent).toBe(
		'Appointments manager'
	)
	expect(screen.getByTestId('app-name').tagName).toBe('H1')

	expect(screen.getByText('Create Appointment')).toBeInTheDocument()
	expect(screen.getByText('No Appointments')).toBeInTheDocument()
})

it('<App /> - Creating appointment and dynamic title', () => {
	render(<App />)

	//1st appointment
	userEvent.type(screen.getByTestId('pet'), 'my pet')
	userEvent.type(screen.getByTestId('owner'), 'rpxic')
	userEvent.type(screen.getByTestId('date'), '2019-01-10')
	userEvent.type(screen.getByTestId('time'), '12:00')
	userEvent.type(screen.getByTestId('symptoms'), 'No drink')

	//click button
	const button = screen.getByTestId('btn-submit')
	userEvent.click(button)

	//2nd appointment
	// userEvent.type(screen.getByTestId('pet'), 'my pet2')
	// userEvent.type(screen.getByTestId('owner'), 'rpxic')
	// userEvent.type(screen.getByTestId('date'), '2019-01-10')
	// userEvent.type(screen.getByTestId('time'), '12:00')
	// userEvent.type(screen.getByTestId('symptoms'), 'No drink')

	// userEvent.click(button)

	//the alert should not appear
	const alert = screen.queryByTestId('alert')
	expect(alert).not.toBeInTheDocument()

	//dynamic title
	expect(screen.getByTestId('dynamic-title').textContent).toBe(
		'Manage Your Appointments'
	)
	expect(screen.getByTestId('dynamic-title').textContent).not.toBe(
		'No Appointments'
	)
})

it('<App /> - Appointments', async () => {
	render(<App />)

	const appointments = await screen.findAllByTestId('appointment')

	expect(appointments).toMatchSnapshot()

	expect(screen.getByTestId('delete-btn')).toBeInTheDocument()
	expect(screen.getByTestId('delete-btn').tagName).toBe('BUTTON')

	expect(screen.getByText('my pet')).toBeInTheDocument()
})

it('<App /> - Delete appointment', async () => {
	render(<App />)

	const deleteBtn = screen.getByTestId('delete-btn')
	expect(deleteBtn).toBeInTheDocument()
	expect(deleteBtn.tagName).toBe('BUTTON')

	userEvent.click(deleteBtn)
	expect(deleteBtn).not.toBeInTheDocument()
	expect(screen.queryByText('my pet')).not.toBeInTheDocument()
	expect(screen.queryByTestId('appointment')).not.toBeInTheDocument()
})
