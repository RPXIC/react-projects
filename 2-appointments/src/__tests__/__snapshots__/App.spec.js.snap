// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> - Appointments 1`] = `
Array [
  <div
    class="appointment"
    data-testid="appointment"
  >
    <p>
      Pet: 
      <span>
        my pet
      </span>
    </p>
    <p>
      owner: 
      <span>
        rpxic
      </span>
    </p>
    <p>
      Date: 
      <span>
        2019-01-10
      </span>
    </p>
    <p>
      Time: 
      <span>
        12:00
      </span>
    </p>
    <p>
      Symptoms: 
      <span>
        No drink
      </span>
    </p>
    <button
      class="button delete u-full-width"
      data-testid="delete-btn"
    >
      Delete Ã—
    </button>
  </div>,
]
`;
